# Variable	Description
- subject_id--> Subject ID
- COVID --> COVID status (tested positive prior to enrollment or during hospitalization)    0 = negative    1= positive
- Age cat--> Age category    1 = 20-34    2 = 36-49    3 = 50-64    4 = 65-79    5 = 80+ 
- BMI cat--> Body mass index:    0 = <18.5 (underweight)    1 = 18.5-24.9 (normal)    2 = 25.0-29.9 (overweight)    3 = 30.0-39.9 (obese)    4 = >=40 (severely obese)    5 = Unknown
- HEART--> Pre-existing heart disease – HEART - (coronary artery disease, congestive heart failure, valvular disease)    0 = No    1 = Yes
- LUNG--> Pre-existing lung disease – LUNG - (asthma, COPD, requiring home O2, any chronic lung condition)    0 = No    1 = Yes    
- KIDNEY--> Pre-existing kidney disease – KIDNEY - (chronic kidney disease, baseline creatinine >1.5, ESRD)    0 = No    1 = Yes    
- DIABETES--> Pre-existing diabetes – DIABETES - (pre-diabetes, insulin and non-insulin dependent diabetes)    0 = No    1 = Yes    
- HTN-->Pre-existing hypertension - HTN    0 = No    1 = Yes    
- IMMUNO--> Pre-existing immunocompromised condition – IMMUNO (active cancer, chemotherapy, transplant, immunosuppressant agents, aspenic)    0 = No    1 = Yes    
- Resp_Symp--> Respiratory symptoms – Symp_Resp (sore throat, congestion, productive or dry cough, shortness of breath or hypoxia, or chest pain)    0 = No    1 = Yes    
- Fever_Sympt--> Febrile symptom
- GI_Symp--> Any GI related symptoms at presentation (abdominal pain, nausea, vomiting, diarrhea)
- D0_draw--> Study draw on Day 0   0 = No   1 = Yes
- D3_draw--> Study draw in Day 3 window (study day 2 to study day 4)   0 = No   1 = Yes
- D7_draw--> Study draw in Day 7 window (study day 5 to study day 9)   0 = No   1 = Yes
- DE_draw--> Study draw for an event - event can include decompensation such as ICU admit or intubation, or extubation   0 = No   1 = Yes
- Acuity 0--> Acuity score maximum for day 0 study window - highest Acuity within Day 0 window (enrollment plus 24 hours) :    1 = Death     2 = Intubated / ventilated      3 = Hospitalized, supplementary O2 required     4 = Hospitalized, no supplementary O2 required     5 = Discharged / Not hospitalized
- Acuity 3--> Acuity score maximum for day 3 study window (study day 2 to study day 4):    1 = Death     2 = Intubated / ventilated      3 = Hospitalized, supplementary O2 required     4 = Hospitalized, no supplementary O2 required     5 = Discharged / Not hospitalized
- Acuity 7--> Acuity score maximum for day 7 study window (study day 5 to study day 9):    1 = Death     2 = Intubated / ventilated      3 = Hospitalized, supplementary O2 required     4 = Hospitalized, no supplementary O2 required     5 = Discharged / Not hospitalized
- Acuity 28--> Acuity score on day 28:   1 = Death     2 = Intubated / ventilated      3 = Hospitalized, supplementary O2 required     4 = Hospitalized, no supplementary O2 required     5 = Discharged / Not hospitalized
- Acuity max--> Acuity max is the highest Acuity level between Day 0 -28     1 = Death within 28 days     2 = Intubated / ventilated, survived to 28 days     3 = Hospitalized, supplementary O2 required, survived to 28 days     4 = Hospitalized, no supplementary O2 required, survived to 28 days     5 = Discharged, was not admitted to hospital within 28 day window, survived
- abs_neut_0_cat--> Absulute neutrophil count day 0 category:    1 = 0-0.99    2 = 1.0-3.99    3 = 4.0-7.99    4 = 8.0-11.99    5 = 12+
- abs_lymph_0_cat--> Absolute lymphocyte count day 0 category:    1 = 0-0.49    2 = 0.50-0.99    3 = 1.00-1.49    4 = 1.50-1.99    5 = 2+    
- abs_mono_0_cat--> Absolute monocyte day 0 category    1 = 0-0.24    2 = 0.25-0.49    3 = 0.50-0.74    4 = 0.75-0.99    5 = 1.0+
- creat_0_cat--> Creatinine day 0 category    1 = 0-0.79    2 = 0.80-1.19    3 = 1.20-1.79    4 = 1.80-2.99    5 = 3+
- crp_0_cat--> c-reactive protein day 0 category:    1 = 0-19.9    2 = 20-59.0    3 = 60-99.9    4 = 100-179    5 = 180+
- ddimer_0_cat--> D-dimer  day 0 category:    1 = 0-499    2 = 500-999    3 = 1000-1999    4 = 2000-3999    5 = 4000+
- ldh_0_cat--> Lactate dehydrogenase day 0 category:    1 = 0-200    2 = 200-299    3 = 300-399    4 = 400-499    5 = 500+
- Trop_72h--> Cardiac event – Trop_72h - (hs-cTn =>100 within first 72 hours of presentation)    0 = No    1 = Yes    
- abs_neut_3_cat--> Absulute neutrophil count day 3 category:    1 = 0-0.99    2 = 1.0-3.99    3 = 4.0-7.99    4 = 8.0-11.99    5 = 12+
- abs_lymph_3_cat--> Absolute lymphocyte count day 3 category:    1 = 0-0.49    2 = 0.50-0.99    3 = 1.00-1.49    4 = 1.50-1.99    5 = 2+    
- abs_mono_3_cat--> Absolute monocyte count day 3 category:    1 = 0-0.24    2 = 0.25-0.49    3 = 0.50-0.74    4 = 0.75-0.99    5 = 1.0+
- creat_3_cat--> Creatinine day 3 category    1 = 0-0.79    2 = 0.80-1.19    3 = 1.20-1.79    4 = 1.80-2.99    5 = 3+
- crp_3_cat--> c-reactive protein day 3 category:    1 = 0-19.9    2 = 20-59.0    3 = 60-99.9    4 = 100-179    5 = 180+
- ddimer_3_cat--> D-dimer  day 3 category:    1 = 0-499    2 = 500-999    3 = 1000-1999    4 = 2000-3999    5 = 4000+
- ldh_3_cat--> Lactate dehydrogenase day 3 category:    1 = 0-200    2 = 200-299    3 = 300-399    4 = 400-499    5 = 500+
- abs_neut_7_cat--> Absulute neutrophil count day 7 category:    1 = 0-0.99    2 = 1.0-3.99    3 = 4.0-7.99    4 = 8.0-11.99    5 = 12+    
- abs_lymph_7_cat--> Absolute lymphocyte count day 7 category:    1 = 0-0.49    2 = 0.50-0.99    3 = 1.00-1.49    4 = 1.50-1.99    5 = 2+    
- abs_mono_7_cat--> Absolute monocyte count day 7 category:    1 = 0-0.24    2 = 0.25-0.49    3 = 0.50-0.74    4 = 0.75-0.99    5 = 1.0+
- creat_7_cat--> Creatinine day 7 category    1 = 0-0.79    2 = 0.80-1.19    3 = 1.20-1.79    4 = 1.80-2.99    5 = 3+
- crp_7_cat--> c-reactive protein day 7 category:    1 = 0-19.9    2 = 20-59.0    3 = 60-99.9    4 = 100-179    5 = 180+
- ddimer_7_cat--> D-dimer  day 3 category:    1 = 0-499    2 = 500-999    3 = 1000-1999    4 = 2000-3999    5 = 4000+
- ldh_7_cat--> 	Lactate dehydrogenase day 7 category:    1 = 0-200    2 = 200-299    3 = 300-399    4 = 400-499    5 = 500+

# Libraries

In [8]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px
import sklearn as svn
%matplotlib inline

from PIL import Image
import plotly.graph_objects as go
from plotly.subplots import make_subplots
import plotly.figure_factory as ff
from prettytable import PrettyTable
from PIL import Image, ImageDraw, ImageFont

# Loading dataset

In [5]:
data = pd.read_csv('MGH_COVID_Clinical_Info.txt', sep=';')

# Brief Information

In [3]:
from pandas_profiling import ProfileReport

In [6]:
data.head(2)

subject_id  COVID  Age_cat  BMI_cat  HEART  LUNG  KIDNEY  DIABETES  HTN  \
0           1      1        1        4      0     0       0         0    0   
1           2      1        2        2      0     0       0         0    0   

   IMMUNO  ...  crp_3_cat  ddimer_3_cat  ldh_3_cat  abs_neut_7_cat  \
0       0  ...        1.0           1.0        1.0             NaN   
1       0  ...        2.0           NaN        NaN             NaN   

   abs_lymph_7_cat  abs_mono_7_cat  creat_7_cat  crp_7_cat  ddimer_7_cat  \
0              NaN             NaN          NaN        NaN           NaN   
1              NaN             NaN          NaN        NaN           NaN   

   ldh_7_cat  
0        NaN  
1        NaN  

[2 rows x 44 columns]

**Data was loaded correctly**

In [9]:
beautitable(data)
print('Table-I. Description of variables in clinical covid dataset')

+-----------------+-----------+---------+-----+--------+---------+--------+-----+-----+
|     Variable    | Data_type | Non-NaN | NaN | Unique | Example | Median | Max | Min |
+-----------------+-----------+---------+-----+--------+---------+--------+-----+-----+
|    subject_id   |   int64   |   384   |  0  |  384   |    1    | 193.5  | 386 |  1  |
|      COVID      |   int64   |   384   |  0  |   1    |    1    |  1.0   |  1  |  0  |
|     Age_cat     |   int64   |   384   |  0  |   5    |    1    |  3.0   |  5  |  1  |
|     BMI_cat     |   int64   |   384   |  0  |   5    |    4    |  2.0   |  5  |  0  |
|      HEART      |   int64   |   384   |  0  |   1    |    0    |  0.0   |  1  |  0  |
|       LUNG      |   int64   |   384   |  0  |   1    |    0    |  0.0   |  1  |  0  |
|      KIDNEY     |   int64   |   384   |  0  |   1    |    0    |  0.0   |  1  |  0  |
|     DIABETES    |   int64   |   384   |  0  |   1    |    0    |  0.0   |  1  |  0  |
|       HTN       |   int64   | 

# Data Cleaning

## Identification of NaN values

In [32]:
data.isna().sum()

subject_id           0
COVID                0
Age_cat              0
BMI_cat              0
HEART                0
LUNG                 0
KIDNEY               0
DIABETES             0
HTN                  0
IMMUNO               0
Resp_Symp            0
Fever_Sympt          0
GI_Symp              0
D0_draw              1
D3_draw              1
D7_draw              1
DE_draw              1
Acuity_0             0
Acuity_3             0
Acuity_7             0
Acuity_28            0
Acuity_max           0
abs_neut_0_cat       4
abs_lymph_0_cat      5
abs_mono_0_cat       8
creat_0_cat          3
crp_0_cat           18
ddimer_0_cat        26
ldh_0_cat           22
Trop_72h             0
abs_neut_3_cat     106
abs_lymph_3_cat    107
abs_mono_3_cat     110
creat_3_cat        105
crp_3_cat          156
ddimer_3_cat       175
ldh_3_cat          191
abs_neut_7_cat     205
abs_lymph_7_cat    206
abs_mono_7_cat     206
creat_7_cat        193
crp_7_cat          240
ddimer_7_cat       251
ldh_7_cat  

## Para este estudio es importante identificar los elementos que afectan a la prognosis del pacientes. Para ello el estudio se centra en las caracteristicas clinicas a t0 (D0_Draw=1)

In [36]:
##droping the only one NaN in D0_draw
df = data.drop(383)

In [40]:
## Generation of dataframe with individuals that were COVID19 positive and started the study at day 0
df=df.loc[(df['COVID']==1)&(df['D0_draw']==1)]

In [59]:
df2 = df.dropna(subset=['abs_neut_0_cat','abs_lymph_0_cat',
                        'abs_mono_0_cat','creat_0_cat',
                        'crp_0_cat','ddimer_0_cat',
                       'ldh_0_cat','ldh_0_cat'])

In [60]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 281 entries, 0 to 382
Data columns (total 44 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   subject_id       281 non-null    int64  
 1   COVID            281 non-null    int64  
 2   Age_cat          281 non-null    int64  
 3   BMI_cat          281 non-null    int64  
 4   HEART            281 non-null    int64  
 5   LUNG             281 non-null    int64  
 6   KIDNEY           281 non-null    int64  
 7   DIABETES         281 non-null    int64  
 8   HTN              281 non-null    int64  
 9   IMMUNO           281 non-null    int64  
 10  Resp_Symp        281 non-null    int64  
 11  Fever_Sympt      281 non-null    int64  
 12  GI_Symp          281 non-null    int64  
 13  D0_draw          281 non-null    float64
 14  D3_draw          281 non-null    float64
 15  D7_draw          281 non-null    float64
 16  DE_draw          281 non-null    float64
 17  Acuity_0        

**Hay un total de 281 individuos que son elegibles para este estudio**

In [61]:
df2.to_csv('COVID_pos.csv')

# Annex
**Functions for making personalized graphs and smart analysis** 

## Beautiful table

In [1]:
##Generation of a beauti-table
def beautitable(data):
    table = PrettyTable(['Variable',
                         'Data_type',
                         'Non-NaN',
                         'NaN',
                         'Unique',
                         'Example',
                        'Median',
                        'Max',
                        'Min'])
    for i in data.columns:
        table.add_row([i, 
                       data[i].dtype,
                       len(data[i]) - np.sum(data[i].isnull()),
                       np.sum(data[i].isnull()),
                       np.count_nonzero(data[i].unique()),
                       data[~data[i].isnull()][i].iloc[0],
                       data[i].median(),
                       data[i].max(),
                       data[i].min()
                    ])
    print(table)
    print()
    return           